controllers/workflow_controller.go

```diff
@@ -0,0 +1,152 @@
+type WorkflowReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=dws.cray.hpe.com,resources=workflows,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=dws.cray.hpe.com,resources=workflows/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=dws.cray.hpe.com,resources=workflows/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Workflow object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *WorkflowReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	log := r.Log.WithValues("Workflow", req.NamespacedName)
+	log.Info("Reconciling Workflow")
+
+	// Fetch the Workflow workflow
+	workflow := &dwsv1alpha1.Workflow{}
+	if err := r.Get(ctx, req.NamespacedName, workflow); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Nothing to do
+	if workflow.Status.Ready {
+		return ctrl.Result{}, nil
+	}
+
+	// Transitioning states. Nothing to do
+	if workflow.Status.State != workflow.Spec.DesiredState {
+		return ctrl.Result{}, nil
+	}
+
+	// The workflow is being deleted. Nothing to do
+	if !workflow.GetDeletionTimestamp().IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	log.Info("Reconciling Workflow Driver Statuses")
+	desiredState := workflow.Spec.DesiredState
+	directives := workflow.Spec.DWDirectives
+
+	// Create a status updater that handles the call to r.Update() if any of the fields
+	// in workflow.Status{} change. This is necessary since Status is not a subresource
+	// of the workflow.
+	statusUpdater := updater.NewStatusUpdater[*dwsv1alpha1.WorkflowStatus](workflow)
+	defer func() { err = statusUpdater.CloseWithUpdate(ctx, r, err) }()
+
+	// Check workflow for test driver entries
+	for driverStatusIndex, driverStatus := range workflow.Status.Drivers {
+
+		// Skip driverStatus entries of other drivers
+		if DRIVERID != driverStatus.DriverID {
+			continue
+		}
+
+		// Skip driverStatus entries that aren't relevant
+		if desiredState != driverStatus.WatchState {
+			continue
+		}
+
+		// Skip driverStatus entries that have already completed
+		if driverStatus.Completed {
+			continue
+		}
+
+		// Skip driverStatus entries with recorded errors
+		if dwsv1alpha1.StatusError == driverStatus.Status {
+			continue
+		}
+
+		var directive = directives[driverStatus.DWDIndex]
+		args, err := dwdparse.BuildArgsMap(directive)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("Could not parse driver args from directive: %s\n", directive))
```

Comment:
You're using a log object that supports structured logging.  So change all of your log messages, both Error() and Info() to something like:
```
log.Error(err, "basic message", "key1", val1, "key2", val2)
```
The pieces you put into key/val are things you want to easily parse from the log.

See `vendor/github.com/go-logr/logr/README.md` and `vendor/github.com/go-logr/logr/logr.go`.